### ISSUE TITLE
Unable to pair

### ISSUE THREAD

**Issue by thejjmiller** [postlund/pyatv#129]
I am unable to pair to my AppleTV. I have include the debug logs below:

Any ideas?

```
atvremote pair --login_id 00000000-0cd3-6715-fe4c-965137d9454c --address Bedroom.local --debug
Use pin 1234 to pair with "pyatv" (press ENTER to stop)
Using pairing guid: 0xCCE7C43F4EF57A0F
Note: If remote does not show up, try rebooting your Apple TV
DEBUG: Started pairing web server at port 58174
DEBUG: Published zeroconf service: ServiceInfo(type='_touch-remote._tcp.local.', name='0000000000000000000000000000003232235841._touch-remote._tcp.local.', address=b'\xc0\xa8\x01A', port=58174, weight=0, priority=0, server='0000000000000000000000000000003232235841._touch-remote._tcp.local.', properties={b'DvNm': 'pyatv', b'RemV': b'10000', b'DvTy': b'iPod', b'RemN': b'Remote', b'txtvers': b'1', b'Pair': 'CCE7C43F4EF57A0F'})
DEBUG: Published zeroconf service: ServiceInfo(type='_touch-remote._tcp.local.', name='0000000000000000000000000000003232260865._touch-remote._tcp.local.', address=b'\xc0\xa8c\x01', port=58174, weight=0, priority=0, server='0000000000000000000000000000003232260865._touch-remote._tcp.local.', properties={b'DvNm': 'pyatv', b'RemV': b'10000', b'DvTy': b'iPod', b'RemN': b'Remote', b'txtvers': b'1', b'Pair': 'CCE7C43F4EF57A0F'})

DEBUG: Shutting down pairing server
No response from Apple TV!
```

**Comment by postlund** [comment]
There's really no need specifying `--login_id` or `--address` as those parameters are not used when pairing. You can specify `--pairing-guid` if you want to pair using a specific id for some reason instead of a randomly created (which is usaully fine). So `atvremote --debug pair` is sufficient.

Anyway, it looks like the device does not try to connect to pyatv. Did a `pyatv` remote appear in the remote control menu? What is the IP-address of your Apple TV? Looks like zeroconf is published to 192.168.1.10 and 192.168.12.1, so it must be on one of those networks. Also, can you see it with `atvremote scan`?

**Comment by postlund** [comment]
Any update on this?

**Comment by postlund** [comment]
@mmm117 I have not tested on latest tvOS, I will try as soon as I get the chance to do so. But I cannot imagine that the pairing process has changed. The only issue I can see is something network related since zeronconf and binding to ports on various interfaces is involved.

**Comment by mmm117** [comment]
@postlund I checked other AppleTV libs on Github, they all failed since tvOS 11.4..

**Comment by postlund** [comment]
@mmm117 I have tested now as well and you are right, it doesn't work. The hash calculation has probably changed, I will have to look into that. A log from an attempt I made:

```
INFO: Got pairing request from 689135CD3B29254FEC47A13418DCA9FB5D5A21CF with code 3ca681936ad230452439815ded9015f3
DEBUG: Got code 3ca681936ad230452439815ded9015f3, expects 40c7db7703c400bb8bed89a12cdf0518
INFO: 10.0.10.81 - - [21/Aug/2018:17:31:35 +0000] "GET /pair?pairingcode=3CA681936AD230452439815DED9015F3&servicename=689135CD3B29254FEC47A13418DCA9FB5D5A21CF HTTP/1.1" 500 168 "-" "tvcontrold (unknown version) CFNetwork/902.2 Darwin/17.7.0"
```

The fun part is that it's the client, i.e. pyatv in this case, that verifies the device. So it's possible to modify the pairing server to ignore the PIN and always accept the pairing. I tried it out and it works. It's not the correct solution but it's a workaround.

**Comment by postlund** [comment]
An interesting observation that I can see right now is that the same hash is sent from the device no matter what pairing guid is used (or other field in the bonjour service I change) for the same PIN. So either some other identifier is used or the algorithm is constant for the same pin (which would be really stupid).

**Comment by mmm117** [comment]
Sorry, I don't know much about this process, do you think you can handle it?

**Comment by macfloripa** [comment]
How can I modify to ignore the PIN?

**Comment by postlund** [comment]
> How can I modify to ignore the PIN?

I just release 0.3.11 that natively supports bypassing the PIN code. Just run specify none as PIN code:

`atvremote -p none pair`

**Comment by postlund** [comment]
I've made a few fixes regarding pairing (#161 and #164), so this should be fixed now.
